### ISSUE TITLE
Access Policy cannot be changed from Credentialed / Restricted to Restricted / Open

### ISSUE THREAD

**Issue by briangow** [MIT-LCP/physionet-build#1758]
Once a project is submitted as Credentialed / Restricted it cannot be downgraded to Restricted / Open . Attempts to do so will result in an "Invalid submission. See errors below." message. No additional information is provided below. This will occur if a downgrade is attempted both for the editor in the copyedit stage or for the author in a revision stage. 

The error when trying to downgrade from Restricted to Open is:

```
self._errors
{'dua': ['This field is required.'], 'required_trainings': ['This field is required.']}
```

This should be fixed so that editors and authors can change the access policy as desired throughout the publication process.

**Comment by superryeti** [comment]
Weirdly, when I tested locally, 
1. i was able to reproduce the error with `dua` only.( i tried to change accesspolicy from Credentialed --> Restricted, Credentialed --> open, Restricted --> Open)
2. I could reproduce this before project submission, during revision and during copyedit(so basically everywhere)
3. i could only reproduce this error message `Select a valid choice. That choice is not one of the available choices.`
I spent some time on trying to figure it out, and everything seems to be right

In every test, the input tag for `dua` is disabled by the `AccessMetadataForm` and the required  was also removed from the  `input` tag for `dua` by `AccessMetadataForm` was also removed and the `request.POST` doesn't have any data for `dua` and ` required_training` for open access policies.

the `dua` has a foreign key relation to `project.DUA` , so maybe  when the form is initialized on `project/views.py` and `console/views.py`  with ` access_form = forms.AccessMetadataForm(data=request.POST, instance=project, editable=editable)`

django might be doing some kind of check that is preventing us from adding None to the `dua` because of the existing foreign key relation.


The hacky way to fix this (that i found so far) is that we could just delete the fields in the form instead of disabling them
If we remove the input from the form entirely, Django removed the `dua` in the database as well.

```
  if self.access_policy not in {AccessPolicy.CREDENTIALED, AccessPolicy.CONTRIBUTOR_REVIEW}:
           # -self.fields['required_trainings'].disabled = True
           # - self.fields['required_trainings'].required = False
           # - self.fields['required_trainings'].widget = forms.HiddenInput()
           del self.fields['required_trainings']

        if self.access_policy == AccessPolicy.OPEN:
            # -self.fields['dua'].disabled = True
            # -self.fields['dua'].required = False
            # -self.fields['dua'].queryset = DUA.objects.none()
            # -self.fields['dua'].widget = forms.HiddenInput()
            del self.fields['dua']

```

@tompollard  not sure if this is a good way. I was also not able to reproduce the issue mentioned by @briangow  as the one i could reproduce said `Select a valid choice. That choice is not one of the available choices.`. Maybe someone already fixed the original issue

I did a fresh pull of dev and tested

**Comment by superryeti** [comment]
@tompollard , tagging you to bring this to your plate for feedback

**Comment by bemoody** [comment]
> Once a project is submitted as Credentialed / Restricted

It would be surprising if this depended on whether the project was already *submitted*.  It seems like the same thing happens with unsubmitted projects.

Amit's solution makes sense and I don't think that's any more hacky than the existing code.  We do `del fields[...]` in other places I think.

We probably do want to ensure that open access projects have `dua` set to None.  Not sure if that is always the case currently.

As a general strategy, it might be better not to be so strict in AccessMetadataForm - if license/training/DUA is inconsistent with access policy then just set those fields to none - and flag errors in check_integrity instead.

**Comment by tompollard** [comment]
@briangow do you think you could pick this up?

**Comment by briangow** [comment]
Yes, I added it to my to do list. I want to get one other physionet-build PR opened first though.

**Comment by bemoody** [comment]
fixed by #2073
